Signal/src/ViewControllers/AddToBlockListViewController.m

```diff
@@ -316,20 +319,63 @@ - (void)updateBlockButtonEnabling
 
 - (void)blockedPhoneNumbersDidChange:(id)notification
 {
-    // TODO: Once we have a list of contacts, we should update it here.
+    dispatch_async(dispatch_get_main_queue(), ^{
+        _blockedPhoneNumbers = [_blockingManager blockedPhoneNumbers];
+
+        [self updateContacts];
+    });
 }
 
 - (void)signalRecipientsDidChange:(NSNotification *)notification
 {
-    [self updateContacts];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self updateContacts];
+    });
 }
 
 - (void)updateContacts
 {
-    dispatch_async(dispatch_get_main_queue(), ^{
-        self.contacts = self.contactsManager.signalContacts;
-        [self.contactsTableView reloadData];
-    });
+    OWSAssert([NSThread isMainThread]);
+
+    self.contacts = [self filteredContacts];
+    [self.contactsTableView reloadData];
+}
+
+- (BOOL)isContactBlocked:(Contact *)contact
+{
+    if (contact.parsedPhoneNumbers.count < 1) {
+        return YES;
+    }
+
+    for (PhoneNumber *phoneNumber in contact.parsedPhoneNumbers) {
+        if ([_blockedPhoneNumbers containsObject:phoneNumber.toE164]) {
+            return YES;
+        }
+    }
+
+    return NO;
+}
+
+- (BOOL)isCurrentUserContact:(Contact *)contact
+{
+    for (PhoneNumber *phoneNumber in contact.parsedPhoneNumbers) {
+        if ([[phoneNumber toE164] isEqualToString:[TSAccountManager localNumber]]) {
+            return YES;
+        }
+    }
+
+    return NO;
+}
+
+- (NSArray<Contact *> *_Nonnull)filteredContacts
+{
+    NSMutableArray<Contact *> *result = [NSMutableArray new];
+    for (Contact *contact in self.contactsManager.signalContacts) {
+        if (![self isContactBlocked:contact] && ![self isCurrentUserContact:contact]) {
+            [result addObject:contact];
+        }
+    }
+    return result;
```

Comment:
I never bother with this for an object that a) we construct ourselves locally and b) is only used internally within a class - especially a small, simple class like this.  There's no chance we'll ever cast it to mutable and we'll never expose it to anyone else.  Copying unnecessarily is expensive.  Perhaps that's a premature optimization.